pkg/inframanager/store/storesetup.go

```diff
@@ -110,13 +110,33 @@ func SetHostInterfaceMac(mac string) bool {
 		return false
 	}
 
+	if len(ip) == 0 {
+		log.Errorf("Invalid mac address %s", mac)
+		return false
+	}
+	if net.ParseIP(ip) == nil {
+		log.Errorf("Invalid ip address %s", ip)
+		return false
+	}
+
 	setupMutex.Lock()
-	Setup.HostInterfaceMac = mac
+	Setup.HostInterface.Ip = ip
+	Setup.HostInterface.Mac = mac
 	setupMutex.Unlock()
 
 	return true
 }
 
+func SetDefaultRule() {
```

Comment:
Is there anywhere where we reset the defaultrule as well? I am thinking of reboot and edge case scenarios where we should be clearing state variables

Reply:
We are using a boolean variable to maintain the state. Initially the value is false, which mean the rule is not set. During a crash or reboot, these entries are automatically flushed into the store. The inframanager reads the data after the reboot and gets the current status. I don't see any other edge cases as this is done during the init and store should take care of the old entries. Let me know if you find any other cases.